source/server/connection_handler_impl.h

```diff
@@ -136,16 +138,34 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler,
      */
     void newConnection(Network::ConnectionSocketPtr&& socket);
 
+    ActiveConnections& getOrCreateActiveConnections(const Network::FilterChain& filter_chain);
+
     ConnectionHandlerImpl& parent_;
     Network::ListenerPtr listener_;
     const std::chrono::milliseconds listener_filters_timeout_;
     const bool continue_on_listener_filters_timeout_;
     std::list<ActiveTcpSocketPtr> sockets_;
-    std::list<ActiveTcpConnectionPtr> connections_;
+    std::unordered_map<const Network::FilterChain*, ActiveConnectionsPtr> connections_by_context_;
```

Comment:
I think in the future the key here might need to either be a shared_ptr or a weak_ptr, right? Who is going to own the filter chain itself and destruct it when no connections refer to it anymore?

Reply:
Currently it's the threadLocal FilterChainManager, which own the tree-structured lookup table, owns the every single FilterChain.
Every new connection won't increase the refcount. 

> either be a shared_ptr or a weak_ptr
Because of the ownership described above, as long as all connections need to be destructed prior to the tree-structured-lookup-table, we could use rawptr/ref here and below